src/main/java/utils/DatabaseConnector.java

```diff
@@ -42,4 +45,65 @@ public static boolean canConnectToDatabase() {
 			return false;
 		}
 	}
+
+	public static ArrayList<Email> getMailbox() {
+		ArrayList<Email> mailbox = new ArrayList<Email>();
+		try (Connection conn = getConnection()) {
+			String query = "SELECT * FROM MAILBOX";
+
+			try (PreparedStatement ps = conn.prepareStatement(query);
```

Comment:
Put PreparedStatement under the first try catch's resource. 
Something like:
```java
String query = "SELECT * FROM MAILBOX";
try (Connection conn = getConnection();
     PreparedStatement ps = conn.prepareStatement(query)) {
    ...
}
Maybe you can also put the ResultSet within the same resource? If it's possible.
```

Reply:
stack tracing would be harder, no?

Reply:
Stack tracing wouldn't really be any harder, as if you understand what prepareStatement() method throws (can be seen in the documentations). The exceptions are all SQLExceptions.